bigtable/google/cloud/bigtable/cluster.py

```diff
@@ -47,35 +42,97 @@ class Cluster(object):
+                                 Possible values are represented by the
+                                 following constants:
+                                 :data:`google.cloud.bigtable.enums.StorageType.SSD`.
+                                 :data:`google.cloud.bigtable.enums.StorageType.SHD`,
+                                 Defaults to
+                                 :data:`google.cloud.bigtable.enums.StorageType.UNSPECIFIED`.
     """
 
-    def __init__(self, cluster_id, instance, location_id,
-                 serve_nodes=DEFAULT_SERVE_NODES,
-                 default_storage_type=_STORAGE_TYPE_UNSPECIFIED):
+    def __init__(self,
+                 cluster_id,
+                 instance,
+                 location_id=None,
+                 state=None,
+                 serve_nodes=None,
+                 default_storage_type=None):
         self.cluster_id = cluster_id
         self._instance = instance
-        self.location = location_id
+        self.location_id = location_id
+        self.state = state
         self.serve_nodes = serve_nodes
         self.default_storage_type = default_storage_type
 
+    @classmethod
+    def from_pb(cls, cluster_pb, instance):
+        """Creates an cluster instance from a protobuf.
+
+        :type cluster_pb: :class:`instance_pb2.Cluster`
+        :param cluster_pb: An instance protobuf object.
+
+        :type instance: :class:`google.cloud.bigtable.instance.Instance`
+        :param instance: The instance that owns the cluster.
+
+        :rtype: :class:`Cluster`
+        :returns: The Cluster parsed from the protobuf response.
+        :raises: :class:`ValueError <exceptions.ValueError>` if the cluster
+                 name does not match
+                 ``projects/{project}/instances/{instance_id}/clusters/{cluster_id}``
+                 or if the parsed instance ID does not match the istance ID
+                 on the client.
+                 or if the parsed project ID does not match the project ID
+                 on the client.
+        """
+        match_cluster_name = _CLUSTER_NAME_RE.match(cluster_pb.name)
+        if match_cluster_name is None:
+            raise ValueError('Cluster protobuf name was not in the '
+                             'expected format.', cluster_pb.name)
+        if match_cluster_name.group('instance') != instance.instance_id:
+            raise ValueError('Instance ID on cluster does not match the '
+                             'instance ID on the client')
+        if match_cluster_name.group('project') != instance._client.project:
+            raise ValueError('Project ID on cluster does not match the '
+                             'project ID on the client')
+        cluster_id = match_cluster_name.group('cluster_id')
+        location_id = cluster_pb.location.split('/')[-1]
+
+        result = cls(cluster_id, instance, location_id, cluster_pb.state,
+                     cluster_pb.serve_nodes, cluster_pb.default_storage_type)
```

Comment:
Please use keyword arguments for all optional parameters (protecting against order changes like the one I requested above).